code/controllers/globals.dm

```diff
@@ -0,0 +1,69 @@
+GLOBAL_REAL(GLOB, /datum/controller/global_vars)
+
+/datum/controller/global_vars
+	name = "Global Variables"
+
+	var/list/gvars_datum_protected_varlist
+	var/list/gvars_datum_in_built_vars
+	var/list/gvars_datum_init_order
+
+/datum/controller/global_vars/New()
+	if(GLOB)
+		CRASH("Multiple instances of global variable controller created")
+	GLOB = src
+
+	var/datum/controller/exclude_these = new
+	gvars_datum_in_built_vars = exclude_these.vars + list("gvars_datum_protected_varlist", "gvars_datum_in_built_vars", "gvars_datum_init_order")
+	qdel(exclude_these)
+
+	log_to_dd("[vars.len - gvars_datum_in_built_vars.len] global variables")
+
+	Initialize()
+
+/datum/controller/global_vars/Destroy(force)
```

Comment:
this got changed to:

```
/datum/controller/global_vars/Destroy()
	//fuck off kevinz
	return QDEL_HINT_IWILLGC
```

I smell TG Drama @MrStonedOne

Reply:
I don't see any real reason to change it to that, this cleans itself up.

Reply:
kevinz has a tendency to ~~ab~~use proc call and sdql2 in interesting ways.